Source/WebCore/page/Navigation.cpp

```diff
@@ -343,6 +383,25 @@ bool Navigation::hasEntriesAndEventsDisabled() const
     return false;
 }
 
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#resolve-the-finished-promise
+void Navigation::resolveFinishedPromise(const NavigationAPIMethodTracker& apiMethodTracker)
+{
+    // FIXME: ASSERT(apiMethodTracker.committedToEntry);
+    if (apiMethodTracker.committedToEntry) {
+        apiMethodTracker.committedPromise->resolve<IDLInterface<NavigationHistoryEntry>>(*apiMethodTracker.committedToEntry);
+        apiMethodTracker.finishedPromise->resolve<IDLInterface<NavigationHistoryEntry>>(*apiMethodTracker.committedToEntry);
+    }
+    cleanupAPIMethodTracker(apiMethodTracker);
+}
+
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#notify-about-the-committed-to-entry
+static void notifyCommittedToEntry(NavigationAPIMethodTracker& apiMethodTracker, RefPtr<NavigationHistoryEntry> entry)
```

Comment:
The entry param likely violates the rule Chris mentioned: "We only pass smart pointers to transfer ownership".